From dbc1dc99957bef3748cf8205ccf487b898b7af14 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Mon, 15 Jul 2013 15:40:17 +0300
Subject: [PATCH 0833/1825] alp: Added print to U-Boot which indicates if DLB
 is enabled

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit d783a222d4c658ad1cdc19048c862d27e5a9040f

	- Added mvCtrlIsDLBEnabled routine, to Read DLB status register (0x1700, Bit 0)
	- Added print to U-Boot to indiacte if DLB is  enabled/disabled

Change-Id: I10dc6c4ad78f518a581bec11642ebe25beb7dbb7
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/2642
Reviewed-by: Kosta Zertsekel <konszert@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c        |   20 ++++++++++++++++++++
 .../avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h        |    1 +
 2 files changed, 21 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
index 855b399..913d24b 100755
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
@@ -1633,6 +1633,26 @@ MV_U32 ctrlSizeRegRoundUp(MV_U32 size, MV_U32 alignment)
 }
 
 /*******************************************************************************
+ * mvCtrlIsDLBEnabled - Read DLB configuration
+ *
+ * DESCRIPTION: return True if DLB is enabled
+ *
+ * INPUT: None
+ *
+ * OUTPUT: None
+ *
+ * RETURN: MV_TRUE, if DLB is enabled
+ ******************************************************************************/
+MV_BOOL mvCtrlIsDLBEnabled(MV_VOID)
+{
+	MV_U32 reg;
+
+	reg = MV_REG_READ(REG_STATIC_DRAM_DLB_CONTROL);
+
+	return (reg & 0x1) ? MV_TRUE : MV_FALSE;
+}
+
+/*******************************************************************************
 * mvCtrlIsBootFromNOR
 *
 * DESCRIPTION:
diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
index 207e617..d5eab12 100755
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
@@ -330,6 +330,7 @@ MV_STATUS ctrlWinWithinWinTest(MV_ADDR_WIN *pAddrWin1, MV_ADDR_WIN *pAddrWin2);
 MV_VOID mvCtrlSmiMasterSet(MV_SMI_CTRL smiCtrl);
 MV_VOID mvCtrlPwrClckSet(MV_UNIT_ID unitId, MV_U32 index, MV_BOOL enable);
 MV_BOOL mvCtrlPwrClckGet(MV_UNIT_ID unitId, MV_U32 index);
+MV_BOOL mvCtrlIsDLBEnabled(MV_VOID);
 MV_BOOL mvCtrlIsBootFromNOR(MV_VOID);
 MV_BOOL mvCtrlIsBootFromSPI(MV_VOID);
 MV_BOOL mvCtrlIsBootFromNAND(MV_VOID);
-- 
1.7.5.4

